### DISCUSSION TITLE
Interactive get 3d coordinates of the mouse click position.

### DISCUSSION THREAD

**Discussion by prokwas** [pyvista/pyvista#5606] [Q&A] [answered]
Hi, I'm trying to implement interactive point selection in a 3D scene similar to your example (https://github.com/pyvista/pyvista-support/issues/239#issuecomment-725234467).
I get a strange result: if the click is far away from the sphere, the point is selected correctly, but if the click is close to the sphere or on the sphere, the point is "magnetized" to the grid.
How can I make the click result predictable and not depend on the position of the click relative to the 3D model?
I am grateful for the answer in advance!
<img width="512" alt="Снимок экрана 2024-02-11 в 20 23 31" src="https://github.com/pyvista/pyvista/assets/72496926/48141cd5-9330-4699-90d5-1b29d516f716">

Here is my code: 
```

import pyvista as pv
import numpy as np

p = pv.Plotter(notebook=0)
p.add_mesh(pv.Sphere(), show_edges=True)
p.camera.parallel_projection = True

def Mycallback(pos):
    A = np.array(pos)
    p.add_text(f'Click_position: {pos}', name='pos')
    B = np.array(p.camera.direction)
    p.add_text('\n  Camera:'+str(B), name="cpos")
    ray = pv.Line(A, 3*B+A)
    p.add_mesh(ray, color="blue", line_width=5, label="Ray Segment")
    point = pv.Sphere(radius=0.02, center=A, theta_resolution=5, phi_resolution=5)
    p.add_mesh(point, color="red", )
    p.update()


p.track_click_position(callback=Mycallback)
p.show()
```

**Comment by banesullivan** [comment, marked as answer]
You may want to use one of our picking methods for this instead of tracking the mouse position which I think might be in screen space, not world/scene space

- https://docs.pyvista.org/version/stable/examples/02-plot/surface-picking.html
- https://docs.pyvista.org/version/stable/api/plotting/_autosummary/pyvista.Plotter.enable_element_picking.html#pyvista.Plotter.enable_element_picking
- https://docs.pyvista.org/version/stable/api/plotting/_autosummary/pyvista.plotter.enable_point_picking

**Reply by prokwas** [reply]
Thanks to Bane's kind reply [link.](https://github.com/pyvista/pyvista/discussions/5606#discussioncomment-8436194) I figured out how to get the coordinates of a point on a surface on click. This is implemented via "enable_surface_point_picking". Now I can get the 3D coordinates of the click point.
Question: can I get the face number on my triangulated surface and the barycentric coordinates in it, or does this need to be calculated?

**Reply by banesullivan** [reply]
I think the element picking with the "face" mode will track the face index but I'm not sure about barycentric coordinates

**Reply by prokwas** [reply]
Thank you Bane, I really appreciate your prompt help. Unfortunately, I haven't figured out how to make mouse click (picking) events not conflict with each other. If you know how to solve such an issue, I would be grateful for your advice.
